pkg/utils/array/join.go

```diff
@@ -0,0 +1,22 @@
+package array
+
+import (
+	"fmt"
+	"strings"
+)
+
+func CommaJoin[T fmt.Stringer](ts []T) string {
+	return Join(ts, ", ")
+}
+
+func Join[T fmt.Stringer](ts []T, join string) string {
+	return fmt.Sprintf("[%s]", strings.Join(Cast(ts), join))
+}
+
+func Cast[T fmt.Stringer](ts []T) []string {
+	strs := []string{}
+	for _, t := range ts {
+		strs = append(strs, t.String())
+	}
+	return strs
+}
```

Comment:
Yes it is a `Map`, but it is mapping `typecast = (T) => T.String()`, so I called this `Cast`.
I heard that Go is very bad at FP, so I tried to avoid using any functor names i.e. `map`, `filter`, `reduce`. 

That said, 'll make this change anyway.